test/unit/modules/modules.spec.ts

```diff
@@ -209,3 +209,56 @@ test("Export Equals", () => {
 
     expect(result).toBe(true);
 });
+
+const reassignmentTestCases = [
+    "x = true",
+    "[x] = [true]",
+    "[[x]] = [[true]]",
+    "({ x } = { x: true })",
+    "({ y: x } = { y: true })",
+    "({ x = false } = { x: undefined })",
+];
+
+test.each(reassignmentTestCases)("export specifier with reassignment afterwards (%p)", reassignment => {
+    util.testModule`
+            let x: any = false;
+            export { x };
+            ${reassignment};
+        `.expectToMatchJsResult();
+});
+
+test.each(reassignmentTestCases)("export specifier fork (%p)", reassignment => {
+    util.testModule`
+        let x: any = false;
+        export { x as a };
+        export { x as b };
+        ${reassignment};
+    `.expectToMatchJsResult();
+});
+
+test.each(["(() => { let a = 2; a = 3 })()", "{ let a = 2; a = 3 }"])(
+    "does not export shadowed identifiers",
+    shadow => {
+        util.testModule`
+            export let a = 1;
+            ${shadow};
+        `.expectToMatchJsResult();
+    }
+);
+
+test("export as specifier shouldn't effect local vars", () => {
+    util.testModule`
+        let x = false;
+        export { x as a };
+        let a = 5;
+        a = 6;
+    `.expectToMatchJsResult();
+});
+
+test("special export specifier side effects", () => {
+    util.testModule`
+        let x = [0, 1, 2];
+        export { x as a };
+        [x.length] = [0];
```

Comment:
Is that test really required? There's no any special behavior there and it works just because `x` is a reference.